src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/indentation/InputLambda1.java

```diff
@@ -0,0 +1,178 @@
+package com.puppycrawl.tools.checkstyle.indentation; //indent:0 exp:0
+
+import java.util.ArrayList; //indent:0 exp:0
+import java.util.List; //indent:0 exp:0
+import java.util.Map; //indent:0 exp:0
+import java.util.StringJoiner; //indent:0 exp:0
+import java.util.stream.Collector; //indent:0 exp:0
+import java.util.stream.Collectors; //indent:0 exp:0
+import java.util.stream.IntStream; //indent:0 exp:0
+import java.util.stream.Stream; //indent:0 exp:0
+
+public class InputLambda1 { //indent:0 exp:0
+
+  interface Printer //indent:2 exp:2
+  { //indent:2 exp:2
+    void print(String s); //indent:4 exp:4
+  } //indent:2 exp:2
+
+  void function1(Runnable x) { //indent:2 exp:2
+    Runnable r1 = () -> { //indent:4 exp:4
+      x.run(); //indent:6 exp:6
+    }; //indent:4 exp:4
+
+    Runnable r1 = () -> { //indent:4 exp:4
+       x.run(); //indent:7 exp:6 warn
+     }; //indent:5 exp:4 warn
+
+    Runnable r2 = () -> { x.run(); }; //indent:4 exp:4
+
+    Runnable r3 = () -> //indent:4 exp:4
+        x.run(); //indent:8 exp:8
+
+    Runnable r4 = () -> x.run(); //indent:4 exp:4
+
+    Printer r5 = s -> System.out.print(s); //indent:4 exp:4
+
+    Printer r6 = s -> System.out //indent:4 exp:4
+        .print(s); //indent:8 exp:8
+
+    Object o = new Thread(() -> { //indent:4 exp:4
+      x.run(); //indent:6 exp:6
+    }); //indent:4 exp:4
+
+    Runnable r11 = //indent:4 exp:4
+        () -> { //indent:8 exp:8
+          x.run(); //indent:10 exp:10
+        }; //indent:8 exp:8
+
+    Runnable r21 = //indent:4 exp:4
+        () -> { x.run(); }; //indent:8 exp:8
+
+    Runnable r31 = //indent:4 exp:4
+        () -> x //indent:8 exp:8
+            .run(); //indent:12 exp:12
+
+    Runnable r41 = //indent:4 exp:4
+        () -> x.run(); //indent:8 exp:8
+
+    Printer r51 = //indent:4 exp:4
+        s -> System.out.print(s); //indent:8 exp:8
+
+    Printer r61 = //indent:4 exp:4
+        s -> System.out //indent:8 exp:8
+            .print(s); //indent:12 exp:12
+
+    Object o1 = new Thread( //indent:4 exp:4
+        () -> { //indent:8 exp:8
+          x.run(); //indent:10 exp:10
+        }); //indent:8 exp:8
+
+    SomeInterface i1 = (LongTypeName //indent:4 exp:4
```

Comment:
I don't see any declaration of `SomeInterface` - please make input compilable in Java 8.